tests/unit/test_validate_checkpoint_operator.py

```diff
@@ -66,3 +70,119 @@ def configure_checkpoint(context: AbstractDataContext) -> Checkpoint:
         # assert
         assert serialized_result["success"]
         json.dumps(serialized_result)  # result must be json serializable
+
+    def test_context_type_ephemeral(self, mocker: MockerFixture):
+        """Expect that param context_type creates an EphemeralDataContext."""
+        # arrange
+        context_type: Literal["ephemeral"] = "ephemeral"
+
+        def configure_checkpoint(context: AbstractDataContext) -> Checkpoint:
+            return Mock()
+
+        mock_gx = Mock()
+        mocker.patch.dict("sys.modules", {"great_expectations": mock_gx})
+        validate_checkpoint = GXValidateCheckpointOperator(
+            task_id="validate_checkpoint",
+            configure_checkpoint=configure_checkpoint,
+            context_type=context_type,
+        )
+
+        # act
+        validate_checkpoint.execute(context={})
+
+        # assert
+        mock_gx.get_context.assert_called_once_with(mode=context_type)
+
+    def test_context_type_cloud(self, mocker: MockerFixture):
+        """Expect that param context_type creates a CloudDataContext."""
+        # arrange
+        context_type: Literal["cloud"] = "cloud"
+
+        def configure_checkpoint(context: AbstractDataContext) -> Checkpoint:
+            return Mock()
+
+        mock_gx = Mock()
+        mocker.patch.dict("sys.modules", {"great_expectations": mock_gx})
+        validate_checkpoint = GXValidateCheckpointOperator(
+            task_id="validate_checkpoint_success",
+            configure_checkpoint=configure_checkpoint,
+            context_type=context_type,
+        )
+
+        # act
+        validate_checkpoint.execute(context={})
+
+        # assert
+        mock_gx.get_context.assert_called_once_with(mode=context_type)
+
+    def test_context_type_filesystem(self, mocker: MockerFixture):
+        """Expect that param context_type defers creation of data context to user."""
+        # arrange
+        context_type: Literal["file"] = "file"
+
+        def configure_checkpoint(context: AbstractDataContext) -> Checkpoint:
+            return Mock()
+
+        def configure_file_data_context() -> FileDataContext:
+            return Mock()
+
+        mock_gx = Mock()
+        mocker.patch.dict("sys.modules", {"great_expectations": mock_gx})
+        validate_checkpoint = GXValidateCheckpointOperator(
+            task_id="validate_checkpoint_success",
+            configure_checkpoint=configure_checkpoint,
+            context_type=context_type,
+            configure_file_data_context=configure_file_data_context,
+        )
+
+        # act
+        validate_checkpoint.execute(context={})
+
+        # assert
+        mock_gx.get_context.assert_not_called()
```

Comment:
Seems like we should be asserting that we call configure_file_data_context and something is done with it.